File: game_roulette.py
# game_roulette.py
import random
import asyncio
from db_manager import get_user_points, update_user_points

COMMAND = "روليت"

async def handle(bot, message):
    if not message.text.startswith(COMMAND):
        return

    uid = message.from_user.id
    user_points = await get_user_points(uid)

    # ───────── 1. التحقق من الرهان ─────────
    parts = message.text.split(" ")
    if len(parts) < 2:
        await bot.send_message(message.chat.id, "⚠️ يجب تحديد مبلغ الرهان\n💡 مثال: روليت 50")
        return

    try:
        bet = int(parts[1])
    except:
        await bot.send_message(message.chat.id, "❌ الرهان يجب أن يكون رقماً صحيحاً")
        return

    if bet <= 0:
        await bot.send_message(message.chat.id, "🚫 لا يمكنك الرهان بمبلغ صفر أو أقل")
        return

    if bet > user_points:
        await bot.send_message(message.chat.id, f"💸 رصيدك الحالي: {user_points} نقاط\n❌ لا تملك نقاط كافية")
        return

    # ───────── 2. واجهة اللعب ─────────
    start_text = f"""
┏━━━━━━━ ● ━━━━━━━┓
      🎰 روليت الإمبراطورية 🎰
┗━━━━━━━ ● ━━━━━━━┛

💰 الرهان: {bet} نقاط
🌀 تدوير عجلة القدر...
    """
    status_msg = await bot.send_message(message.chat.id, start_text)

    await asyncio.sleep(1.5)
    await bot.edit_message_text(message.chat.id, status_msg.message_id, start_text + "\n\n⚡️ العجلة تتباطأ...")
    await asyncio.sleep(1.5)

    # ───────── 3. تحديد النتيجة ─────────
    outcome = ["win", "lose", "jackpot"]
    weights = [45, 50, 5]

    def weighted_random(items, weights):
        total = sum(weights)
        r = random.uniform(0, total)
        for i, w in enumerate(weights):
            if r < w:
                return items[i]
            r -= w
        return items[-1]

    result = weighted_random(outcome, weights)

    # ───────── 4. النتائج ─────────
    new_points = user_points
    result_text = ""

    if result == "win":
        new_points = await update_user_points(uid, bet)
        result_text = f"""
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ فـوز إمـبـراطـوري ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ الفوز ]
🏆 النقاط المكتسبة : +{bet}
✨ رصيدك الحالي : {new_points} نقاط
        """
    elif result == "jackpot":
        jackpot = bet * 5
        new_points = await update_user_points(uid, jackpot)
        result_text = f"""
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ جاكبوت أسطوري ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ الجاكبوت ]
🏆 النقاط المكتسبة : +{jackpot}
✨ رصيدك الحالي : {new_points} نقاط
        """
    else:  # lose
        new_points = await update_user_points(uid, -bet)
        result_text = f"""
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ خسارة ساحقة ⌯
┗━━━━━━━ ● ━━━━━━━┛

💀 الحظ : [ الخسارة ]
💸 خسارتك : -{bet} نقاط
✨ رصيدك الحالي : {new_points} نقاط
        """

    await bot.edit_message_text(message.chat.id, status_msg.message_id, result_text)

File: game_roulette.js
// game_roulette.js
const db = require('./db_manager');

const COMMAND = "روليت";
const MISSION_TYPE = "roulette";
const REWARD_ITEM = "🎁 صندوق الحظ النادر";

async function handle(bot, ctx) {
    if (!ctx.message.text.startsWith(COMMAND)) return;

    const uid = ctx.from.id;
    const userGold = await db.get_user_gold(uid);

    // ───────── 1. التحقق من الرهان ─────────
    const parts = ctx.message.text.split(" ");
    if (parts.length < 2) {
        return ctx.reply(
            "⚠️ يجب تحديد مبلغ الرهان\n💡 مثال: روليت 500"
        );
    }

    const bet = parseInt(parts[1]);
    if (isNaN(bet)) return ctx.reply("❌ الرهان يجب أن يكون رقماً صحيحاً");
    if (bet <= 0) return ctx.reply("🚫 لا يمكنك الرهان بمبلغ صفر أو أقل");
    if (bet > userGold) return ctx.reply(`💸 رصيدك الحالي: ${userGold} ذهب\n❌ لا تملك ذهباً كافياً`);

    // ───────── 2. واجهة اللعب ─────────
    let startText = `
┏━━━━━━━ ● ━━━━━━━┓
      🎰 روليت الإمبراطورية 🎰
┗━━━━━━━ ● ━━━━━━━┛

💰 الرهان: ${bet} ذهبة
🌀 تدوير عجلة القدر...
    `;
    let statusMsg = await ctx.reply(startText);

    await new Promise(r => setTimeout(r, 1500));
    await bot.editMessageText(ctx.chat.id, statusMsg.message_id, startText + "\n\n⚡️ العجلة تتباطأ...");
    await new Promise(r => setTimeout(r, 1500));

    // ───────── 3. تحديد النتيجة ─────────
    const outcome = ["win", "lose", "jackpot"];
    const weights = [45, 50, 5];

    function weightedRandom(items, weights) {
        const sum = weights.reduce((a, b) => a + b, 0);
        let r = Math.random() * sum;
        for (let i = 0; i < items.length; i++) {
            if (r < weights[i]) return items[i];
            r -= weights[i];
        }
    }

    const result = weightedRandom(outcome, weights);

    // ───────── 4. النتائج ─────────
    let resultText = "";
    let newBal = userGold;

    if (result === "win") {
        newBal = await db.update_user_gold(uid, bet);
        resultText = `
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ فـوز إمـبـراطـوري ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ الفوز ]
💰 الجائزة : +${bet} ذهب
✨ رصيدك الحالي : ${newBal} ذهب
        `;
    } else if (result === "jackpot") {
        const jackpot = bet * 5;
        newBal = await db.update_user_gold(uid, jackpot);
        resultText = `
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ جاكبوت أسطوري ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ الجاكبوت ]
💰 الجائزة : +${jackpot} ذهب
✨ رصيدك الحالي : ${newBal} ذهب
        `;
    } else {
        newBal = await db.update_user_gold(uid, -bet);
        resultText = `
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ خسارة ساحقة ⌯
┗━━━━━━━ ● ━━━━━━━┛

💀 الحظ : [ الخسارة ]
💸 خسارتك : -${bet} ذهب
✨ رصيدك الحالي : ${newBal} ذهب
        `;
    }

    await bot.editMessageText(ctx.chat.id, statusMsg.message_id, resultText);

    // ───────── 5. التحقق من المهمة اليومية ─────────
    const missionCompleted = await db.complete_mission(uid, MISSION_TYPE);
    if (missionCompleted) {
        await db.add_to_inventory(uid, REWARD_ITEM);
        await ctx.reply(`✅ تم إكمال المهمة اليومية!\n🎁 تم إضافة ${REWARD_ITEM} إلى مخزونك`);
    }
}

module.exports = { handle };
